projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon-anchor/cart-coupon-anchor.service.ts

```diff
@@ -0,0 +1,16 @@
+import { EventEmitter, Injectable } from '@angular/core';
```

Comment:
What about renaming this service to:
`CartCouponService.show()' `CartCouponService.getShowEvent()' and moving it into cart coupon component?

Is it that it's cart anchor that want to notify coupon component to show (or to scroll into view in this implementation), not the other way around? If yes, this service belongs more logically to coupon component.

Reply:
Hi, Stan
Yes, the cart anchor is used to scroll into the view of coupon applying area. According to UX design which is also agreed on Bill and Kevin, the anchor should be shown under the 'Proceed to checkout' button, it's different slot with CartCouponComponent, that's why there is a separate component for anchor.
Regarding the CartCouponAnchorService, we did it refer to address-book. The service is different with traditional service, it does nothing about ngrx, so we didn't put the method in CartVoucherService together.

Reply:
I get the idea from UX perspective and agree with it 100%.
I'm referring more to proper service responsibility, but I noticed that my proposition was indeed wrongly named, I should propose `CartCouponComponentService`. 
The idea is to:
1. Make it more logically correct, it's the anchor that wants coupon component to do something, where coupon is master -> so anchor should use coupon component service to control it, not the other way around.
2. Customizability/Extensibility: I can easily imagine that coupon components can exist without an actual anchor component, could be part of the checkout step, could use a different mechanism to show up or we can even have two different "anchor" like components to display it. That's why I believe it should not have the anchor component as a dependency (though it's service).

Reply:
Done